lib/_tls_wrap.js

```diff
@@ -247,7 +247,19 @@ function TLSSocket(socket, options) {
   // starting the flow of the data
   this.readable = true;
   this.writable = true;
-  this.read(0);
+
+  // Don't read until user-land had a chance to setup listeners
+  var self = this;
+  process.nextTick(function() {
+    // Socket already has some buffered data - emulate receiving it
+    if (socket && socket._readableState && socket._readableState.length) {
+      var buf;
+      while ((buf = socket.read()) !== null)
+        self._handle.receive(buf);
+    }
+
+    self.read(0);
+  });
```

Comment:
Awesome, thanks for the tip. I'll add that once we come to a resolution with your previous comment (doing nextTick in constructor and _init or only once).